src/mpid/ch4/netmod/ofi/ofi_nic.h

```diff
@@ -0,0 +1,195 @@
+            fi_freeinfo(MPIDI_OFI_global.prov_use[i]);
+        }
+        MPIDI_OFI_global.num_nics = MPIR_CVAR_CH4_OFI_MAX_NICS;
+    }
+
+    /* Now go through every NIC and set initial information
+     * from current process's perspective */
+    for (int i = 0; i < MPIDI_OFI_global.num_nics; ++i) {
+        nics[i].nic = MPIDI_OFI_global.prov_use[i];
+        nics[i].id = i;
+        /* Determine NIC's "closeness" to current process */
+        nics[i].close = is_nic_close(nics[i].nic);
+        if (nics[i].close)
+            MPIDI_OFI_global.num_close_nics++;
+        /* Set the preference of all NICs to least preferable (lower is more preferable) */
+        nics[i].prefer = MPIDI_OFI_global.num_nics + 1;
+        nics[i].count = 0;
+        nics[i].num_close_ranks = 0;
+        /* Determine NIC's first normal parent topology
+         * item (e.g., typically the socket parent) */
+        nics[i].parent = get_nic_parent(nics[i].nic);
+        /* Expand list of close NIC-parent topology items or increment */
+        if (nics[i].close) {
+            int found = 0;
+            for (int j = 0; j < num_parents; ++j) {
+                if (parents[j] == nics[i].parent) {
+                    found = 1;
+                    break;
+                }
+            }
+            if (!found) {
+                parents[num_parents] = nics[i].parent;
+                num_parents++;
+            }
+        }
+    }
+
+    /* If there were zero NICs on my socket, then just consider every NIC close
+     * and share them among all ranks with a similar view */
+    if (MPIDI_OFI_global.num_close_nics == 0) {
+        for (int i = 0; i < MPIDI_OFI_global.num_nics; ++i)
+            nics[i].close = 1;
+        MPIDI_OFI_global.num_close_nics = MPIDI_OFI_global.num_nics;
+    }
+
+    /* Sort the NICs array based on closeness first. This way all the close
+     * NICs are at the beginning of the array */
+    qsort(nics, MPIDI_OFI_global.num_nics, sizeof(nics[0]), compare_nics);
+
+    /* Because we cannot communicate with the other local processes to avoid collisions with the
+     * same NICs, just shift NICs that have multiple close NICs around according to their local
+     * rank. This will likely give the same result as long as processes have been bound properly. */
+    int old_idx = (MPIDI_OFI_global.num_close_nics == 0) ? 0 :
+        local_rank % MPIDI_OFI_global.num_close_nics;
+
+    if (old_idx != 0) {
+        MPIDI_OFI_nic_info_t *old_nics = MPL_malloc(sizeof(MPIDI_OFI_nic_info_t) *
+                                                    MPIDI_OFI_global.num_nics, MPL_MEM_ADDRESS);
+        memcpy(old_nics, nics, sizeof(MPIDI_OFI_nic_info_t) * MPIDI_OFI_global.num_nics);
+
+        /* Rotate the preferred NIC for each process starting at old_idx. */
+        for (int new_idx = 0; new_idx < MPIDI_OFI_global.num_close_nics; new_idx++) {
+            if (new_idx != old_idx)
```

Comment:
This condition is always true and is unnecessary.

Reply:
Yep. Should have removed this before.